amaranth_boards/tang_nano_9k.py

```diff
@@ -0,0 +1,135 @@
+import os
+import subprocess
+
+from amaranth.build import *
+from amaranth.vendor.gowin import *
+
+try:
+    from .resources import *
+except:
+    from resources import *
+
+
+__all__ = ["TangNano9kPlatform"]
+
+
+class TangNano9kPlatform(GowinPlatform):
+    part          = "GW1NR-LV9QN88PC6/I5"
+    family        = "GW1NR-9C"
+    default_clk   = "clk27" # or "OSC" to use on-chip oscillator
+    osc_frequency = 2500000 # Hz
```

Comment:
Our general policy here is that if `default_clk` points to a pin (like here), then `osc_frequency` is unspecified. However if the external clock is missing or unusable (like on the original Tang Nano), then `default_clk` would be `OSC` and the frequency would be specified by the designer, with some reasonable default (like 24 MHz).